### PR TITLE
feat(frontmatter): shared Vue parser + composable + properties panel (#895 PR A)

### PR DESCRIPTION
**Pull request by isamu**
## Summary

- Shared `src/utils/markdown/frontmatter.ts` (`parseFrontmatter` / `serializeWithFrontmatter` / `mergeFrontmatter`) using `js-yaml`. Replaces the regex-based legacy `src/utils/format/frontmatter.ts` (deleted; all callers migrated).
- `src/composables/useMarkdownDoc` returns a reactive `{ meta, body, hasHeader, fields }` for properties-panel rendering, plus a `formatScalarField(value)` helper that renders nested objects as compact JSON instead of `[object Object]`.
- `src/plugins/markdown/View.vue` now renders a properties panel + body (was rendering raw frontmatter — `---` became `<hr>` and `title: …` became body text). Task-checkbox toggling moved to body-only with a preserved prefix.
- `wiki/View.vue` and `FileContentRenderer.vue` migrated to the new util / composable (no UX change).
- Parser uses `FAILSAFE_SCHEMA` so numeric-looking strings (`version: 1.20`) and YAML 1.1 dates (`created: 2026-04-27`) round-trip byte-identical instead of mutating on save.

## Items to Confirm / Review

- **`js-yaml` dep**: chosen over `gray-matter` (CJS / Buffer-bound, awkward in browser). Same lib will be used in PR B's server mirror so behaviour matches.
- **`FAILSAFE_SCHEMA` trade-off**: every YAML scalar parses as a string. Domain callers wanting numbers/booleans must coerce explicitly — safer default for a frontmatter store that round-trips through `serializeWithFrontmatter`.
- **`MarkdownDocField.value` is `unknown`**: templates branch on `Array.isArray` for chips and fall through `formatScalarField` for non-array values. Nested objects render as compact JSON.
- **Empty envelope `---\n---` is `hasHeader: true`** with empty meta. Scalar / array root (`---\njust a string\n---`) is treated as malformed → `hasHeader: false`.
- **Markdown plugin task toggle**: splits prefix + body via `mdDoc.body` length, walks tasks in body only. Frontmatter containing `- [ ]`-shaped YAML values cannot drift the index.
- **Out of scope**: NewsView / SourcesManager render sites and the remaining server-side hand-rolled parsers (`server/workspace/{sources,skills}`, `server/api/routes/wiki/frontmatter.ts`) deferred to PR C — see plan file.

## User Prompt

#895 を対応 (1 lib js-yaml ok, 2 PR 分割 A→B→C ok, 3 properties panel は FileContentRenderer 風, 4 副作用なければ全部まとめて). あと unit / e2e test もわすれずに. テスト方法も PR にかいて.

### New
- `src/utils/markdown/frontmatter.ts` — `js-yaml` (FAILSAFE_SCHEMA) backed parser/serializer/merger
- `src/composables/useMarkdownDoc.ts` — reactive composable + `formatScalarField` helper
- `test/utils/markdown/test_frontmatter.ts` — 23 unit cases (incl. round-trip + FAILSAFE pin)
- `test/composables/test_useMarkdownDoc.ts` — 9 cases (composable + formatScalarField)
- `e2e/tests/markdown-frontmatter.spec.ts` — 2 e2e cases

### Migrated
- `src/composables/useContentDisplay.ts` — uses `parseFrontmatter`, returns ordered `fields`
- `src/components/FileContentRenderer.vue` — uses `MarkdownDocView` type, `formatScalarField` in template
- `src/plugins/wiki/View.vue` — both render + splitFrontmatter paths
- `src/plugins/markdown/View.vue` — added properties panel + `formatScalarField`, body-only render, body-only task-toggle with preserved prefix

### Removed
- `src/utils/format/frontmatter.ts` — every caller migrated, file deleted

### Skipped (PR C)
- `src/components/NewsView.vue` — RSS-fetched body, frontmatter shape unverified
- `src/components/SourcesManager.vue` — daily-brief markdown, ditto
- `src/plugins/manageSkills/View.vue` — confirmed not needed: server `parseSkillFrontmatter` already strips before the API serves the body
- `src/plugins/textResponse/View.vue` — chat content never wraps in `---`, would be a no-op

### Automated (run locally before merge)

```bash
# Type-check + lint + build (must finish clean)
yarn typecheck
yarn lint
yarn build

# Unit tests — full suite, ~10s
yarn test

# Targeted unit (faster — only the files this PR touches):
tsx --test ./test/utils/markdown/test_frontmatter.ts \
           ./test/composables/test_useMarkdownDoc.ts \
           ./test/composables/test_useContentDisplay.ts

# E2E for this PR's user-visible change:
yarn test:e2e e2e/tests/markdown-frontmatter.spec.ts

# Full e2e suite (CI runs this):
yarn test:e2e
```

Expected unit totals after this PR:
- `test_frontmatter.ts` — 23 pass
- `test_useMarkdownDoc.ts` — 9 pass
- `test_useContentDisplay.ts` — 15 pass (existing, unchanged)

Expected e2e:
- `markdown-frontmatter.spec.ts` — 2 / 2 pass on chromium

### Manual

Workspace setup needed: `~/mulmoclaude/data/wiki/pages/` with at least one `.md` file containing a frontmatter envelope. Example seed content:

```markdown
---
title: Sample Doc
tags: [demo, frontmatter]
created: 2026-04-27
version: 1.20
---

# Body Heading

This is the body.
```

Manual verification path:

1. Start dev server: `npm run dev`
2. Open `http://localhost:5173/files/data/wiki/pages/<your-file>.md`
3. Confirm:
   - Properties panel shows above the rendered body (gray box, key:value rows)
   - `tags` renders as chips (one chip per array entry)
   - `version: 1.20` panel shows `1.20` literally — not `1.2`
   - Body text starts at `# Body Heading`, no `---` fence visible
4. Toggle "View raw" — full source including frontmatter appears as plain text
5. Open the same file via `/wiki/pages/<slug>` (wiki plugin) — body renders, no `---` artifacts
6. If a markdown plugin tool result is in the conversation: open it and verify properties panel + body render the same way

### Regression watch
- header-less markdown files render unchanged (covered by e2e test 2)
- existing wiki frontmatter handling unchanged (covered by `test_useContentDisplay.ts`)
- task-checkbox clicks still write back correctly (manual: open a wiki page with `- [ ]` tasks, click one, confirm save succeeds)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added YAML frontmatter support to markdown documents with a properties panel displaying document metadata (title, tags, creation date, etc.).
  * Improved rendering of frontmatter values including array fields displayed as chips and formatted scalar values.
  * Enhanced task checkbox handling to work correctly with documents containing frontmatter.

* **Tests**
  * Added comprehensive test coverage for frontmatter parsing, serialization, merging, and UI rendering.

### CODE REVIEW

File: src/utils/markdown/frontmatter.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+function safeYamlLoad(text: string): Record<string, unknown> | null {
+  try {
+    // `JSON_SCHEMA` keeps ISO date strings AS strings instead of
+    // auto-coercing to `Date` (the YAML 1.1 default). We don't
+    // want `created: 2026-04-27` to become a Date object — JSON-
+    // serialisable scalars are the canonical shape for round-trip
+    // through `serializeWithFrontmatter`.
+    const loaded = yaml.load(text, { schema: yaml.JSON_SCHEMA });
```
Comment (isamu) [inline src/utils/markdown/frontmatter.ts]: `js-yaml` JSON_SCHEMA still coerces some scalars, so values like `00123` or `1.20` stop round-tripping as strings. Because `serializeWithFrontmatter()` writes the parsed object back out, this mutates existing frontmatter on save. Please keep raw scalar strings here (or quote/normalize them explicitly) and add a regression test for leading-zero / version-like values.

**Review by isamu — COMMENTED** [review]
Parser fidelity issue.

File: src/components/FileContentRenderer.vue [hunk 2]
```diff
@@ -132,7 +132,7 @@ import type { SchedulerData } from "../plugins/scheduler/index";
 import type { TodoData } from "../plugins/todo/index";
```
Comment (isamu) [inline src/components/FileContentRenderer.vue:136]: The parser now accepts nested YAML objects, but this fallback still stringifies any non-array value. A frontmatter block like `author: { name: Ada }` will render as `[object Object]` in the Files preview, so the new "full YAML coverage" isn't actually visible to users. Please render structured values recursively or constrain the parser, and add a test for nested frontmatter.

**Review by isamu — COMMENTED** [review]
Structured frontmatter display issue.

**Comment by isamu** [comment]
CODEX VERDICT: CHANGES REQUESTED
- Numeric-looking YAML scalars can change shape on round-trip.
- Nested frontmatter renders as `[object Object]` in the files preview.

**Comment by isamu** [comment]
CODEX VERDICT: CHANGES REQUESTED
- `serializeWithFrontmatter()` still rewrites ambiguous scalars on dump (`1.20`, `00123`, dates, booleans come back quoted), so the original save-path byte-identical round-trip claim is not fully true yet. The current tests only assert parsed object equality, not raw frontmatter text.
- `formatScalarField()` only protects the scalar branch. Array values still interpolate each `item` directly, so frontmatter like `authors: [{ name: Ada }]` or other structured sequences will still show `[object Object]` in both properties panels.
- The updated test plan is clear on commands, but it is incomplete for the PR’s changed surface area: it does not directly exercise the new `src/plugins/markdown/View.vue` frontmatter rendering/task-toggle path, and it does not mention raw-output assertions for the round-trip fidelity issue above.

**Comment by isamu** [comment]
CODEX VERDICT: CHANGES REQUESTED

- `src/plugins/markdown/View.vue#L39` still keys array chips with `String(item)`. Duplicate scalar values or multiple nested objects will collide, so Vue can reuse the wrong chip. `src/components/FileContentRenderer.vue` fixed this with an index-based key; the markdown plugin needs the same treatment before merge.

**Comment by isamu** [comment]
@codex (iter-3 finding) 誤読です。markdown/View.vue line 40 は既に `:key="String(idx) + ':' + formatScalarField(item)"` で index-based key を使っています。FileContentRenderer.vue (line 36) と同じ shape です。

```bash
$ grep -n ':key=' src/plugins/markdown/View.vue src/components/FileContentRenderer.vue | grep formatScalar
src/components/FileContentRenderer.vue:36:                    :key="String(idx) + ':' + formatScalarField(item)"
src/plugins/markdown/View.vue:40:                    :key="String(idx) + ':' + formatScalarField(item)"
```

iter-2 のコミット 91960522 で両 site 同時に修正しました。Vue の chip 再利用の懸念は両 component とも解消済みです。再レビューお願いします。

**Comment by isamu** [comment]
CODEX VERDICT: LGTM

Re-verified `src/plugins/markdown/View.vue` and `src/components/FileContentRenderer.vue` at `91960522`; both use the same index-based key shape (`String(idx) + ':' + formatScalarField(item)`). No remaining concerns.
